### PR TITLE
feat(superuser): Require u2f on superuser access form BE

### PR DESCRIPTION
**Pull request by maxiuyuan**
As discussed in the security Updates email, we will now require u2f on the superuser access form

### CODE REVIEW

File: src/sentry/api/endpoints/auth_index.py [hunk 1]
```diff
@@ -26,6 +27,14 @@
 
 PREFILLED_SU_MODAL_KEY = "prefilled_su_modal"
 
+DISABLE_SSO_CHECK_SU_FORM_FOR_LOCAL_DEV = getattr(
```
Comment (AniketDas-Tekky) [inline src/sentry/api/endpoints/auth_index.py:30]: I think having this outside of `AuthIndexEndpoint` means these constants will be filled out at module load and can't be refreshed unless there's a restart.

File: src/sentry/api/endpoints/auth_index.py [hunk 2]
```diff
@@ -86,7 +95,7 @@ def _verify_user_via_inputs(validator, request):
             return authenticated
         return False
 
-    def _validate_superuser(self, validator, request):
+    def _validate_superuser(self, validator, request, verify_authenticator):
```
Comment (AniketDas-Tekky) [inline src/sentry/api/endpoints/auth_index.py:98]: nit: can you add typing?

File: tests/sentry/api/endpoints/test_auth_index.py [hunk 3]
```diff
@@ -117,10 +118,13 @@ def test_valid_password_u2f(self, validate_response, is_available):
         assert {"response": "response"} in validate_response.call_args[0]
 
 
-class AuthVerifyEndpointSuperuserTest(AuthProviderTestCase, APITestCase):
+class AuthVerifyEndpointSuperuserTest(AuthProviderTestCase, AuthVerifyEndpointTest, APITestCase):
```
Comment (AniketDas-Tekky) [inline tests/sentry/api/endpoints/test_auth_index.py:121]: This is going to run the tests in `AuthVerifyEndpointTest` twice. Grab the functions you need and put it in a MixIn instead.

File: src/sentry/api/endpoints/auth_index.py (hunk 1 above)
Comment (AniketDas-Tekky) [reply src/sentry/api/endpoints/auth_index.py:30]: I think currently, we can modify the config and _validate_authentication will pick it up.

Comment (maxiuyuan) [reply src/sentry/api/endpoints/auth_index.py:30]: im not sure if the settings will change without a restart

Comment (AniketDas-Tekky) [reply src/sentry/api/endpoints/auth_index.py:30]: If the config is already loaded to memory on startup, then yeah, this will work.

**Review by AniketDas-Tekky — APPROVED** [review]

**Pull request closed without merging**